bitcoin/src/p2p/message.rs

```diff
@@ -537,6 +620,79 @@ impl Decodable for RawNetworkMessage {
+                        payload: Vec::consensus_decode_from_finite_reader(r)?,
+                    },
+                }
+            }
+            1u8 => NetworkMessage::Addr(Decodable::consensus_decode_from_finite_reader(r)?),
+            2u8 => NetworkMessage::Block(Decodable::consensus_decode_from_finite_reader(r)?),
+            3u8 => NetworkMessage::BlockTxn(Decodable::consensus_decode_from_finite_reader(r)?),
+            4u8 => NetworkMessage::CmpctBlock(Decodable::consensus_decode_from_finite_reader(r)?),
+            5u8 => NetworkMessage::FeeFilter(Decodable::consensus_decode_from_finite_reader(r)?),
+            6u8 => NetworkMessage::FilterAdd(Decodable::consensus_decode_from_finite_reader(r)?),
+            7u8 => NetworkMessage::FilterClear,
+            8u8 => NetworkMessage::FilterLoad(Decodable::consensus_decode_from_finite_reader(r)?),
+            9u8 => NetworkMessage::GetBlocks(Decodable::consensus_decode_from_finite_reader(r)?),
+            10u8 => NetworkMessage::GetBlockTxn(Decodable::consensus_decode_from_finite_reader(r)?),
+            11u8 => NetworkMessage::GetData(Decodable::consensus_decode_from_finite_reader(r)?),
+            12u8 => NetworkMessage::GetHeaders(Decodable::consensus_decode_from_finite_reader(r)?),
+            13u8 => NetworkMessage::Headers(
+                HeaderDeserializationWrapper::consensus_decode_from_finite_reader(r)?.0,
+            ),
+            14u8 => NetworkMessage::Inv(Decodable::consensus_decode_from_finite_reader(r)?),
+            15u8 => NetworkMessage::MemPool,
+            16u8 => NetworkMessage::MerkleBlock(Decodable::consensus_decode_from_finite_reader(r)?),
+            17u8 => NetworkMessage::NotFound(Decodable::consensus_decode_from_finite_reader(r)?),
+            18u8 => NetworkMessage::Ping(Decodable::consensus_decode_from_finite_reader(r)?),
+            19u8 => NetworkMessage::Pong(Decodable::consensus_decode_from_finite_reader(r)?),
+            20u8 => NetworkMessage::SendCmpct(Decodable::consensus_decode_from_finite_reader(r)?),
+            21u8 => NetworkMessage::Tx(Decodable::consensus_decode_from_finite_reader(r)?),
+            22u8 => NetworkMessage::GetCFilters(Decodable::consensus_decode_from_finite_reader(r)?),
+            23u8 => NetworkMessage::CFilter(Decodable::consensus_decode_from_finite_reader(r)?),
+            24u8 =>
+                NetworkMessage::GetCFHeaders(Decodable::consensus_decode_from_finite_reader(r)?),
+            25u8 => NetworkMessage::CFHeaders(Decodable::consensus_decode_from_finite_reader(r)?),
+            26u8 =>
+                NetworkMessage::GetCFCheckpt(Decodable::consensus_decode_from_finite_reader(r)?),
+            27u8 => NetworkMessage::CFCheckpt(Decodable::consensus_decode_from_finite_reader(r)?),
+            28u8 => NetworkMessage::AddrV2(Decodable::consensus_decode_from_finite_reader(r)?),
+            _ =>
+                return Err(encode::Error::Parse(encode::ParseError::ParseFailed(
```

Comment:
Ideally this error returned the unknown short ID, but I think this requires a change to the `encode::ParseError` which lives over in the `concensus` module...not sure if we want to touch that API at the moment with all the refactoring.